### ISSUE TITLE
Chef 13 deprecation warning

### ISSUE THREAD

**Issue by kpmueller** [3ofcoins/chef-cookbook-hostname#64]
Using it on chef 12 is producing this warning:

```
Deprecated features used!
  method access to node attributes (node.foo.bar) is deprecated and will be removed in Chef 13, please use bracket syntax (node["foo"]["bar"]) at 1 location:
    - /var/chef/cache/cookbooks/hostname/attributes/default.rb:4:in `from_file'
   See https://docs.chef.io/deprecations_attributes.html for further details.
```

It's not clear to me why it's being produced, though..

**Comment by mburns** [comment]
The version in git (`develop` branch) has diverged significantly from what is posted to the Supermarket. This warning shouldn't show up if you point your Berksfile to `cookbook 'hostname' github: '3ofcoins/chef-cookbook-hostname'` until it gets published.

**Comment by rmoriz** [comment]
So what happend with the original author? Should we fork this?

**Comment by jeffbyrnes** [comment]
I was just coming here to ask that the v0.4.1 and v0.4.2 releases be published to Supermarket, guess this is the place!

**Comment by pronoiac** [comment]
@mburns: From here, if my `Berksfile` includes: 
```
cookbook 'hostname', github: '3ofcoins/chef-cookbook-hostname'
```

`berks update` puts into `Berksfile.lock`:
```
  hostname
    git: https://github.com/3ofcoins/chef-cookbook-hostname.git
    revision: bb2af2424269348ab1d48f0e975b7ae14a6aa969
```

And [that revision](https://github.com/3ofcoins/chef-cookbook-hostname/commit/bb2af2424269348ab1d48f0e975b7ae14a6aa969) dates to August 2015.

**Comment by mburns** [comment]
@pronoiac this repo (that I don't have admin rights over, I'm just a user that was given merge access) uses the `develop` branch as its 'default', but also has a `master` branch. Sorry I left that step out, it is easy to miss.

```
cookbook 'hostname', github: '3ofcoins/chef-cookbook-hostname', branch: 'develop'
```

Alternatively, you could use [`tag: 'v0.4.2'`](https://github.com/3ofcoins/chef-cookbook-hostname/releases/tag/v0.4.2) to lock it to a  version that I've released (but can't upload to the Supermarket).

**Comment by rmoriz** [comment]
Maybe @mpasternacki was unable to recover the Supermarket key? There was a bug in Chef ID/Supermarket until yesterday which returned empty keys. This is fixed now status.chef.io/incidents/4jqvsy99wsr0

**Comment by kpmueller** [comment]
The supermarket version is what's used by default, so can this be fixed?

**Comment by xamebax** [comment]
Hi, I will be looking through all issues and PRs this week (most probably this weekend), including bringing the Supermarket version up to date.

**Comment by xamebax** [comment]
@mburns @rmoriz @mpasternacki @pronoiac @jeffbyrnes @kpmueller I merged the develop branch into master and published the newest version on the Supermarket.

**Comment by jeffbyrnes** [comment]
@xamebax a question on what, exactly you published. Did you publish:

* The current `HEAD` of `master`? (d2e0534)
* Tag [`v0.4.2`](https://github.com/3ofcoins/chef-cookbook-hostname/tree/v0.4.2)

It’s always nicest when the tags/releases here reflect the actual releases published to Supermarket, but I get the impression you published d2e0534.

Thankfully, the [diff between `v0.4.2` and `master`](https://github.com/3ofcoins/chef-cookbook-hostname/compare/v0.4.2...master) doesn’t have any functional code changes, so no real harm done if that’s how it went.

If I may, one more suggestion: if your intent is to not use the `develop` branch any longer, you may want to change this repo’s settings so that `master` is the default branch, and delete `develop`. You can do that in the [repo’s branch settings](https://github.com/3ofcoins/chef-cookbook-hostname/settings/branches).

**Comment by xamebax** [comment]
Yes, I published the current `HEAD` of `master`. I agree, publishing the tag would've been more kosher, sorry about that.

I am not sure if I would personally want to change the current branching model (this model always made most sense to me), but I stopped working for Three of Coins almost two years ago and it's not my decision to make :) I guess that's also why it took such a long time to respond on my side.
